src/test/Semaphore.t.sol

```diff
@@ -1,126 +1,254 @@
+    function testCannotRegisterIfProofIncorrect() public {
+        // Setup
+        Semaphore.MerkleTreeProof memory badProof = Semaphore.MerkleTreeProof(
+            proof.ar,
+            proof.bs,
+            Semaphore.G1Point(
+                0x23115ff1573808639f19724479b195b7894a45c9868242ad2a416767359c6c78,
+                0x23f3fa30273c7f38e360496e7f9790450096d4a9592e1fe6e0a996cb04b8fb28
+            )
+        );
+        bytes memory expectedError =
+            abi.encodeWithSelector(Semaphore.ProofValidationFailure.selector);
+        vm.expectRevert(expectedError);
+
+        // Test
+        semaphore.registerIdentities(badProof, preRoot, startIndex, identityCommitments, postRoot);
+    }
 
-        semaphore.transferAccess(user);
+    /// @notice Checks that it reverts if the provided start index is incorrect.
+    function testCannotRegisterIfStartIndexIncorrect() public {
+        bytes memory expectedError =
+            abi.encodeWithSelector(Semaphore.ProofValidationFailure.selector);
+        vm.expectRevert(expectedError);
 
-        assertEq(semaphore.manager(), user);
+        // Test
+        semaphore.registerIdentities(proof, preRoot, startIndex + 1, identityCommitments, postRoot);
     }
 
-    function testCanCreateGroup() public {
-        uint256 groupId = 0;
-        uint8 depth = 20;
-
-        hevm.expectEmit(true, false, false, true);
-        emit GroupCreated(groupId, depth, 0);
+    /// @notice Checks that it reverts if you pass it an incorrect set of identities.
+    function testCannotRegisterIfIdentitiesIncorrect() public {
+        // Setup
+        identityCommitments[2] = 0x7F;
+        bytes memory expectedError =
+            abi.encodeWithSelector(Semaphore.ProofValidationFailure.selector);
+        vm.expectRevert(expectedError);
 
-        semaphore.createGroup(groupId, depth);
+        // Test
+        semaphore.registerIdentities(proof, preRoot, startIndex, identityCommitments, postRoot);
     }
 
-    function testAddMemberEvent() public {
-        uint256 groupId = 1;
-        uint256 leafIndex = uint256(0);
-
-        semaphore.createGroup(groupId, 20);
+    /// @notice Checks that it reverts if you pass it the wrong post root.
+    function testCannotRegisterIfPostRootIncorrect() public {
+        bytes memory expectedError =
+            abi.encodeWithSelector(Semaphore.ProofValidationFailure.selector);
+        vm.expectRevert(expectedError);
 
-        hevm.expectEmit(true, false, false, true);
-        emit MemberAdded(groupId, leafIndex, identityCommitment, updatedRoot);
+        // Test
+        semaphore.registerIdentities(proof, preRoot, startIndex, identityCommitments, postRoot + 1);
+    }
 
-        semaphore.addMember(groupId, identityCommitment);
+    /// @notice Tests that it reverts if you try and register identities as a non manager.
+    function testCannotRegisterIdentitiesAsNonManager() public {
+        // Setup
+        address prankAddress = address(0);
```

Comment:
Don't use `0`, `0` is special and could conceivably be filtered out for an unrelated reason.